### PR TITLE
expression: round the frac part for `ParseTimeFromFloat64`

### PR DESCRIPTION
**Pull request by gengliqi**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/56339

Problem Summary:
See https://github.com/pingcap/tidb/issues/56339

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by dveeden** [comment]
@mjonss this might be related to (or conflict with) what you are working on.

**Comment by mjonss** [comment]
> @mjonss this might be related to (or conflict with) what you are working on.

Yes, if this is merged, then I need to update #56442 as well.

File: pkg/types/time_test.go [hunk 1]
```diff
@@ -1814,7 +1814,8 @@ func TestParseTimeFromFloat64(t *testing.T) {
 		{0.0, mysql.TypeDate, 0, 0, 0, 0, 0, 0, 0, nil},
 		{20000102030405, mysql.TypeDatetime, 2000, 1, 2, 3, 4, 5, 0, nil},
 		{20000102030405.015625, mysql.TypeDatetime, 2000, 1, 2, 3, 4, 5, 15625, nil},
-		{20000102030405.0078125, mysql.TypeDatetime, 2000, 1, 2, 3, 4, 5, 7812, nil},
+		{20000102030405.0078125, mysql.TypeDatetime, 2000, 1, 2, 3, 4, 5, 7813, nil},
```
Comment (mjonss) [inline pkg/types/time_test.go:1817]: The significant digits of the float fractional part is only '.008' I guess, so does it really matter if it is '.007812' or '.007813' ?
Just try this with your PR:
```
{20000102030405.0078139, mysql.TypeDatetime, 2000, 1, 2, 3, 4, 5, 7814, nil},
```

Comment (gengliqi) [reply pkg/types/time_test.go:1817]: Yes. The fractional part is only `.008` so the fractional part of your case is still `7813`.
As https://github.com/pingcap/tidb/issues/56339#issuecomment-2401552101 said, I just want to make this behavior consistent with TiKV and MySQL.

Comment (mjonss) [reply pkg/types/time_test.go:1817]: OK, then I see.
Can you please add tests that will test TiKV/TiFlash/UniStore handling of co-processor calls for this (if exists), so we can show that all four implementations are the same?

**Comment by dveeden** [comment]
@gengliqi could you fix the conflict?

**Comment by gengliqi** [comment]
> @gengliqi could you fix the conflict?

Fixed. Thanks for reminding me.

File: pkg/types/time_test.go (hunk 1 above)
Comment (dveeden) [reply pkg/types/time_test.go:1817]: @mjonss is this now resolved?

**Review by dveeden — APPROVED** [review]

File: pkg/types/time_test.go (hunk 1 above)
Comment (dveeden) [reply pkg/types/time_test.go:1817]: @gengliqi I only see a single integration test. Are you sure it is testing with both TiKV and Unistore?

Comment (gengliqi) [reply pkg/types/time_test.go:1817]: Yes, the `check_dev` test runs these integration tests with Unistore while the `check_dev_2` runs with a real TiKV.

**Comment by dveeden** [comment]
@gengliqi any idea about why the checks are failing?

**Review by mjonss — APPROVED** [review]

**Pull request merged**